internal/pkg/manifest/env.go

```diff
@@ -69,6 +71,73 @@ type environmentVPCConfig struct {
 	Subnets subnetsConfiguration `yaml:"subnets,omitempty"`
 }
 
+type environmentCDNConfig struct {
+	EnableCDN *bool
+	CDNConfig AdvancedCDNConfig // mutually exclusive with EnableCDN
+}
+
+// AdvancedCDNConfig represents an advanced configuration for a Content Delivery Network.
+type AdvancedCDNConfig struct {
+	PrefixListIngress *bool `yaml:"public_ingress_allowed"`
+}
+
+// IsEmpty returns whether AdvancedCDNConfig is empty.
+func (a *AdvancedCDNConfig) IsEmpty() bool {
```

Comment:
nit: can we rename all Config receivers to
```suggestion
func (cfg *AdvancedCDNConfig) IsEmpty() bool {
```